src/form/widget/widget.js

```diff
@@ -0,0 +1,379 @@
+
+        /**
+         * Gets the value of the widget
+         * @returns {String|String[]}
+         */
+        getValue() {
+            return delegate('getValue');
+        },
+
+        /**
+         * Sets the value of the widget
+         * @param {String|String[]} value
+         * @returns {widgetForm}
+         * @fires change after the value has been changed
+         */
+        setValue(value) {
+            this.getConfig().value = value;
+            delegate('setValue', value);
+            this.notify();
+
+            return this;
+        },
+
+        /**
+         * Gets access to the validation engine
+         * @returns {validator}
+         */
+        getValidator() {
+            return validator;
+        },
+
+        /**
+         * Sets the validation engine
+         * @param {validationRule|validationRule[]|validator} validation
+         * @returns {widgetForm}
+         */
+        setValidator(validation) {
+            if (validation && _.isFunction(validation.validate)) {
+                validator = validation;
+            } else {
+                if (validation && !validation.validations) {
+                    if (!_.isArray(validation)) {
+                        validation = [validation];
+                    }
+                    validation = {
+                        validations: validation
+                    };
+                }
+
+                validator = validatorFactory(validation);
+            }
+
+            return this;
+        },
+
+        /**
+         * Resets the widget to the default validators
+         * @returns {widgetForm}
+         */
+        setDefaultValidators() {
+            // restore factory default validators
+            this.setValidator(this.getConfig().validator);
+
+            // then apply provider default validators
+            delegate('setDefaultValidators');
+
+            return this;
+        },
+
+        /**
+         * Resets the widget to its default value
+         * @returns {widgetForm}
+         */
+        reset() {
+            delegate('reset');
+            setInvalidState(false);
+            return this;
+        },
+
+        /**
+         * Serializes the value of the widget
+         * @returns {widgetValue}
+         */
+        serialize() {
+            return delegate('serialize');
+        },
+
+        /**
+         * Validates the widget
+         * @returns {Promise}
+         */
+        validate() {
+            return this.getValidator()
+                .validate(this.getValue())
+                .then(res => {
+                    setInvalidState(false);
+                    return res;
+                })
+                .catch(err => {
+                    setInvalidState(true, err);
+                    return Promise.reject(err);
+                });
+        },
+
+        /**
+         * Triggers the change event
+         * @returns {widgetForm}
+         * @fires change
+         */
+        notify() {
```

Comment:
I also found this method name too broad : `change` or `update` ? What do you think ?

Reply:
What about `notifyChange` ?

Reply:
I will remove this API, it should remain internal. I though some widget implementation could use it, but this should not be the case.